aiogram_dialog/manager/manager.py

```diff
@@ -177,20 +195,24 @@ async def start(
         if mode is StartMode.NORMAL:
             await self._start_normal(state, data)
         elif mode is StartMode.RESET_STACK:
-            await self.reset_stack(remove_keyboard=False)
+            await self.reset_stack(remove_keyboard=False, update_text=None)
             await self._start_normal(state, data)
         elif mode is StartMode.NEW_STACK:
             await self._start_new_stack(state, data)
         else:
             raise ValueError(f"Unknown start mode: {mode}")
-
-    async def reset_stack(self, remove_keyboard: bool = True) -> None:
+    
+    async def reset_stack(self, remove_keyboard: bool = True, update_text: str = None) -> None:
         self.check_disabled()
         storage = self.storage()
         stack = self.current_stack()
         while not stack.empty():
             await storage.remove_context(stack.pop())
         await storage.save_stack(stack)
+        if remove_keyboard and update_text:
+            await self._update_text(update_text)
+        elif update_text:
+            await self._update_text(new_text=update_text, reset_last_message_id=True)
```

Comment:
i dont understand. Why in case of `update_text` we need to reset stack and do not need in case of `remove_keyboard and update_text`